gammapy/cube/new.py

```diff
@@ -158,7 +158,7 @@ def make_map_exposure_true_energy(pointing, livetime, aeff, ref_geom, offset_max
     # We check if exposure is a 3D array in case there is a single bin in energy
     # TODO: call np.atleast_3d ?
     if len(exposure.shape) < 3:
-        exposure = np.expand_dims(exposure, 0)
+        exposure = np.expand_dims(exposure, 0) * exposure.unit
```

Comment:
It seems weird to have to do this check / handling of 2D case here.
I think it's not present e.g. in the corresponding function that fills a map from background IRF.
@registerrier @AtreyeeS - Are the IRF evaluate methods different? Can they be change to behave the same?

Alternatively, I'd suggest to pass `np.expand_dims(exposure.value, 0)`, because in the future, `np.expand_dims` might be changed to work with Quantity and then this code would have the unit factor twice and break.
So let's always use the pattern to strip off the unit and only pass the value for functions that can't deal with Quantity.